org_fedora_oscap/rule_handling.py

```diff
@@ -668,3 +683,71 @@ def eval_rules(self, ksdata, storage, report_only=False):
             return []
 
     # nothing to be reverted for now
+
+
+class KdumpRules(RuleHandler):
+    """Simple class holding data from the rules affecting the kdump addon."""
+
+    def __init__(self):
+        """Constructor setting the initial value of attributes."""
+
+        self._kdump_enabled = None
+
+    def kdump_enabled(self, kdenabled):
+        """Enable or Disable Kdump"""
+
+        if kdenabled == True:
+            self._kdump_enabled = True
+        if kdenabled == False:
+            self._kdump_enabled = False
+
+    def __str__(self):
+        """Standard method useful for debugging and testing."""
+
+        ret = "kdump"
+
+        if self._kdump_enabled == True:
+            ret += " --enable"
+
+        if self._kdump_enabled == False:
+            ret += " --disable"
+
+        return ret
+
+    def eval_rules(self, ksdata, storage, report_only=False):
+        """:see: RuleHandler.eval_rules"""
+
+        messages = []
+
+        if self._kdump_enabled == None:
+            return []
+
+        if self._kdump_enabled == False:
+            msg = _("Kdump will be disabled on startup")
+        if self._kdump_enabled == True:
+            msg = _("Kdump will be enabled on startup")
+
+        messages.append(RuleMessage(self.__class__,
+                                    common.MESSAGE_TYPE_INFO, msg))
+
+        try:
+            if not report_only:
+                ksdata.addons.com_redhat_kdump.enabled = self._kdump_enabled
+        except AttributeError:
+            log.warning("com_redhat_kdump is not installed. "
+                        "Skipping kdump configuration")
+
+        return messages
+
+    def revert_changes(self, ksdata, storage):
+        """:see: RuleHander.revert_changes"""
+
+        try:
+            if ksdata.addons.com_redhat_kdump.enabled == False or \
+               self._kdump_enabled == False:
```

Comment:
I do not understand this condition. why are you checking both?

Reply:
forgot to remove that